### ISSUE TITLE
Fix user stop bugs in Tower of Hanoi

### ISSUE THREAD

**Issue by JMCanning78** [JMCanning78/datastructures-visualization#155]
There are various ways users can cause stack traces when stopping the animations in the Tower of Hanoi puzzle.

Procedure to cause bug 1:

1. Make a new puzzle with 2 disks
2. Move animation speed to slow end
3. Drag top disk from left spindle away from spindle, but not on top either of the other two spindes
4. Release the top disk and click on the second disk before the animation moves it back to the left spindle.

This causes a stack trace like:
```
Exception in Tkinter callback
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/tkinter/__init__.py", line 1705, in __call__
    return self.func(*args)
  File "TowerOfHanoi.py", line 349, in handler
    self.restoreDisks()
  File "TowerOfHanoi.py", line 377, in restoreDisks
    self.moveItemsTo(items, target, steps=10, sleepTime=0.01)
  File "/Users/canning/Documents/textbooks/data_structures/Python/datastructures-visualization/PythonVisualizations/VisualizationApp.py", line 658, in moveItemsTo
    self.wait(sleepTime)
  File "/Users/canning/Documents/textbooks/data_structures/Python/datastructures-visualization/PythonVisualizations/VisualizationApp.py", line 780, in wait
    raise UserStop()      # animation while waiting then raise exception
VisualizationApp.UserStop
```

Procedure to cause bug 2:

1. Make a new puzzle with 3 disks
2. Start the animation of the solution.
3. Click on a disk while the animation runs.

The animation stops and there is no stack trace, but subsequent clicks on and drags of disks cause problems like lower disks covering up higher disks and spindles not being drawn all the way down to the topmost disk (the internal representation thinks there are more disks on the spindle than are being drawn).

There are other conditions that cause stack traces but they are hard to reproduce.  Try solving the puzzle and stopping during the different kinds of animations.  

There's a distinction between animations that should be interruptible such as those illustrating a data structure operation, and those used to restore the state of items on the canvas after a user stop request.  The restoration animations shouldn't be interruptible.  I'm not sure the best way to implement this distinction.

**Comment by JMCanning78** [comment]
Some of these bugs might be solved by commit e1d663c which was made on the Issue190-create-step-mode branch.

**Comment by JMCanning78** [comment]
Made branch Issue155-fix-user-stop-bugs to work on this.
